src/parser.cc

```diff
@@ -832,6 +852,62 @@ bool readDoc(tinyxml2::XMLDocument *_xmlDoc, ElementPtr _sdf,
   return true;
 }
 
+//////////////////////////////////////////////////
+bool checkXml(tinyxml2::XMLElement *_xml, const std::string &_source,
+    Errors &_errors)
+{
+  // A null XML element is still valid as it might not be a mandatory element.
+  // Further errors will be deciphered by calling readXml with its SDF ptr.
+  if (!_xml)
+    return true;
+
+  std::string errorSourcePath = _source;
+  if (_source == kSdfStringSource || _source == kUrdfStringSource)
+    errorSourcePath = "<" + _source + ">";
+
+  // Top level models must have an empty relative_to frame on the top level
+  // pose.
+  {
+    if (tinyxml2::XMLElement *topLevelElem = _xml->FirstChildElement("model"))
+    {
+      std::unordered_set<std::string> frames;
+      tinyxml2::XMLElement *frame = topLevelElem->FirstChildElement("frame");
+      while (frame)
+      {
+        if (const char* frameName = frame->Attribute("name"))
+        {
+          frames.insert(std::string(frameName));
+        }
+        frame = frame->NextSiblingElement("frame");
+      }
+
+      if (tinyxml2::XMLElement *topLevelPose =
+          topLevelElem->FirstChildElement("pose"))
+      {
+        if (const char *relativeTo = topLevelPose->Attribute("relative_to"))
+        {
+          const std::string relativeToStr(relativeTo);
+          if (!relativeToStr.empty() &&
+              frames.find(relativeToStr) == frames.end())
```

Comment:
I think this is doing a lot more than it should. The top level model should not reference anything in it's `//model/pose/@relative_to` regardless of whether there are frames within it that match. This is because the scope of `//model/pose/@relative_to` is the siblings of `//model`. If there were other frames outside of the model, it would be possible to reference those, but since this is the top level model, we are assured there is nothing outside of the model.

Reply:
Per VC, I corrected the failing test, https://github.com/ignitionrobotics/sdformat/blob/sdf11/test/integration/frame.cc#L43-L52, and just checked for the top level models, without considering frames, https://github.com/ignitionrobotics/sdformat/pull/636/commits/828bd9e46f2c2d4263b259094f73144db5569e61